### ISSUE TITLE
textureProj with shadow fails on MSL

### ISSUE THREAD

**Issue by HansKristian-ARM** [KhronosGroup/SPIRV-Cross#239]
@billhollings please see https://github.com/KhronosGroup/SPIRV-Cross/pull/238/commits/f337e50844e8130a7f9a8944e233233fb138a961.

The error from MSL compiler is https://travis-ci.org/KhronosGroup/SPIRV-Cross/jobs/259275392:
```
shaders-msl/frag/texture-proj-shadow.frag

Reference shader path: reference/shaders-msl/frag/texture-proj-shadow.frag

reference/shaders-msl/frag/texture-proj-shadow.frag:18:52: error: unknown type name 'unknown_depth_texture_type'

fragment main0_out main0(main0_in in [[stage_in]], unknown_depth_texture_type<float> uShadow1D [[texture(0)]], sampler uShadow1DSmplr [[sampler(0)]], depth2d<float> uShadow2D [[texture(1)]], sampler uShadow2DSmplr [[sampler(1)]], texture1d<float> uSampler1D [[texture(2)]], sampler uSampler1DSmplr [[sampler(2)]], texture2d<float> uSampler2D [[texture(3)]], sampler uSampler2DSmplr [[sampler(3)]], texture3d<float> uSampler3D [[texture(4)]], sampler uSampler3DSmplr [[sampler(4)]])
```

**Comment by HansKristian-ARM** [comment]
I disabled that test from MSL for now, but would be good to add it when it works.

**Comment by billhollings** [comment]
Fix with PR #254.
